.github/workflows/buildAndTest.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 The StableHLO Authors.
+
+name: Build and Test
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+  schedule:
+    # Always regenerate once every day
+    - cron:  '* 12 * * *'
+  workflow_dispatch:
+
+# Ensure that only a single job or workflow using the same
+# concurrency group will run at a time. This would cancel
+# any in-progress jobs in the same github workflow and github
+# ref (e.g. refs/heads/main or refs/pull/<pr_number>/merge).
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Build using CMake and run tests.
+# Use Cached LLVM to improve build times.
+# Current setup only builds on linux.
```

Comment:
Also, what about Windows? How hard would it be to implement it as well? This definitely doesn't have to block this CL - can just open a ticket for that if it's realistic, or we can drop this topic if it's not.

Reply:
Would require some new scripts since most of our workflows are bash. None of the other MLIR OSS projects seem to have actions for windows. Still maybe worth considering?

Reply:
"None of the other MLIR OSS projects seem to have actions for windows". Good to know! Let's ignore for now then.